providers/implementations/kdfs/hkdf.c

```diff
@@ -267,9 +290,15 @@ static int hkdf_common_set_ctx_params(KDF_HKDF *ctx, const OSSL_PARAM params[])
     if (ossl_param_is_empty(params))
         return 1;
 
-    if (OSSL_PARAM_locate_const(params, OSSL_ALG_PARAM_DIGEST) != NULL) {
+    if ((p = OSSL_PARAM_locate_const(params, OSSL_ALG_PARAM_DIGEST)) != NULL) {
         const EVP_MD *md = NULL;
 
+        if (ctx->fixed_digest) {
+            ERR_raise_data(ERR_LIB_PROV, PROV_R_DIGEST_NOT_ALLOWED,
+                           "Setting the digest is not supported for fixed-digest HKDFs");
+            return 0;
+        }
```

Comment:
Comparing the digest against existing and allow no-op?

Reply:
This was discussed here and the decision was to just error out if any attempt to set the digest was made: https://github.com/openssl/openssl/pull/27247#discussion_r2034616644

Reply:
Thanks for clarification!